### ISSUE TITLE
Unable to use a Gateway group (LB/FO) for routing Wireguard VPN traffic.

### ISSUE THREAD

**Issue by mibur70** [opnsense/core#6981]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [X ] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [X ] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**
opnsense version 23.7.7_1.
wireguard 2.4_1 Kernel implementation. (i have not tried wireguard-go)

This is a relative new setup.
IPv4 is being used, IPv6 is disabled.

I have created two seperate Wireguard VPN Tunnels (two instance/peer combinations).
When i put their respective gateways into a Gateway group, either Failover or Load balanced and redirect traffic to it, opnense always tries to route through the default gateway.
When i redirect traffic to either VPN gateway, it works just fine.
I also have two OpenVPN tunnels running with a FailOver gateway group and this works just fine.
Settings like sticky connections have no impact, this looks like a routing issue.

**To Reproduce**
1. Create a Wireguard site-to-site tunnel with peer/instance combination.
2. Assign the wg1 VPN to an interface WAN_WG1
3. make sure to have a WAN_WG1_GW (dynamic)
4. create an outbound NAT for WAN_WG1
5. create an outbound FW rule on the LAN interface for routing traffic from a specific IP to any but redirect to gateway WAN_WG1. Set a local tag on this rule NO_VPN_LEAK. Log traffic.
6. create a FW rule on the WAN interface (direction OUT) whitch blocks all traffic that has a local tag set NO_VPN_LEAK. log traffic.
7. confirm that this setup works. when the gateway and/or VPN is down, outbound WAN traffic is blocked.

8. Repeat step 1-7 but now for tunnel wg2, interface WAN_WG2 and Gateway WAN_WG2_GW

9. Create a gateway group WIREGUARD_FO and put gateway WAN_WG1_GW as TIER1 and WAN_WG2_GW as tier2
10. change the FW rule to redirect outbound traffic to WIREGUARD_FO instead of either WAN_WG1_GW or WAN_WG2_GW.
11. Confirm no traffic at all from the source ip's into the tunnel, no icmp, no http(s).
12. confirm that the FW live view shows that the routing rule to WIREGUARD_FO is being hit (due to logging being on).
13. confirm that the FW live view shows that traffic is actually being routed to the default WAN and shows blocked due to local tag NO_VPN_LEAK.

14. Change the LAN outbound rule to redirect either to WAN_GW1_GW or WAN_GW2_GW.
15. confirm that traffic is now being routed properly again.

**Expected behavior**
I would expect to be able to group two working Wireguard VPN gateways into a gateway group and use that for either Load Balancing or Fail Over.

**Describe alternatives you considered**
For now, i fallback to OpenVPN

**Screenshots**

Personally i don't think screenshots will add any additional info. The individual tunnels work fine.
This is my gateway group though;
![afbeelding](https://github.com/opnsense/core/assets/73356185/921eddbe-f7cb-4d73-80bf-0dcda3062c29)
and these are my individual gateways:
![afbeelding](https://github.com/opnsense/core/assets/73356185/c3e6414e-dd15-488e-9302-0a5c9a5ef7af)

![afbeelding](https://github.com/opnsense/core/assets/73356185/21e97d85-78f1-453d-9a07-dd097196133f)

**Relevant log files**

Please let me know which logfiles are relevant. I do have pfsense but not much of opensense.

**Environment**
opnsense version 23.7.7_1.
wireguard 2.4_1 Kernel implementation. (i have not tried wireguard-go)
OPNsense is virtualized in proxmox with virtio network drivers. Hardware is Core-i5 1235u, 64GB mem (VM has 16GB)
All networking including vlan tags are managed by openvswitch on proxmox) with I226 2.5Gbps network interfaces running 1Gbps,
ISP is connected with 1Gbps synchronious PPPOE.

**Comment by spin-lock** [comment]
I had this problem before when I was trying to use dynamic gateway as a member of gateway group.
Current OPNsense does not support dynamic gateway as a member of gateway group (it does not support route by '-net'). 
The solution is to make sure each of member of gateway has an IP address assigned.
Refer to this [https://github.com/opnsense/core/issues/6486](url)

**Comment by AdSchellevis** [comment]
If it's the same as https://github.com/opnsense/core/issues/6486, there indeed is no simple fix to offer this functionality due to validation constraints  (I did try something back then https://github.com/opnsense/core/commit/362b08622a9f7da00cd8e701f421b8c063a44010 , but practically this has a high risk of breaking your ruleset).

We should explain this constraint in the documentation, I've opened a ticket for that https://github.com/opnsense/docs/issues/513. Close this issue?
